fetch/api/response/response-consume.html

```diff
@@ -56,80 +70,134 @@
+    checkResponseBody(urlSearchParams, urlSearchParamsData, "URLSearchParams", "text", checkBodyText);
+    checkResponseBody(urlSearchParams, urlSearchParamsData, "URLSearchParams", "arrayBuffer", checkBodyArrayBuffer);
 
-    function checkReadableStreamResponseBody(streamData, bodyType, checkFunction) {
+    function checkReadableStreamResponseBody(streamData, bodyConsumingType, checkFunction, responseInit) {
       promise_test(function(test) {
         var stream = new ReadableStream({
           start: function(controller) {
             controller.enqueue((stringToArray(streamData)));
             controller.close();
           }
         });
-        var response = new Response(stream);
+        var response = new Response(stream, responseInit);
         assert_false(response.bodyUsed, "bodyUsed is false at init");
         return checkFunction(response, streamData);
-      }, "Consume stream response's body as " + bodyType);
+      }, "Consume stream response's body as " + bodyConsumingType);
     }
 
-    checkReadableStreamResponseBody(textData, "blob", checkBodyBlob);
+    checkReadableStreamResponseBody(textData, "blob", checkBodyBlob, textResponseInit);
     checkReadableStreamResponseBody(textData, "text", checkBodyText);
-    checkReadableStreamResponseBody(textData, "json", checkBodyJSON);
     checkReadableStreamResponseBody(textData, "arrayBuffer", checkBodyArrayBuffer);
+    checkReadableStreamResponseBody(textData, "json", checkBodyJSON);
+    checkReadableStreamResponseBody(formTextData, "formData with correct multipart type", checkBodyFormDataMultipart, formTextResponseInit);
+    checkReadableStreamResponseBody(formTextData, "formData without correct multipart type", checkBodyFormDataError);
+    checkReadableStreamResponseBody(urlSearchParamsData, "formData with correct urlencoded type", checkBodyFormDataUrlencoded, urlSearchParamsResponseInit);
+    checkReadableStreamResponseBody(urlSearchParamsData, "formData without correct urlencoded type", checkBodyFormDataError);
 
-    function checkFetchedResponseBody(bodyType, checkFunction) {
+    function checkFetchedResponseBody(bodyConsumingType, checkFunction) {
       return promise_test(function(test) {
         return fetch("../resources/top.txt").then(function(response) {
           assert_false(response.bodyUsed, "bodyUsed is false at init");
           return checkFunction(response, "top");
         });
-      }, "Consume fetched response's body as " + bodyType);
+      }, "Consume fetched response's body as " + bodyConsumingType);
     }
     checkFetchedResponseBody("blob", checkBodyBlob);
     checkFetchedResponseBody("text", checkBodyText);
     checkFetchedResponseBody("arrayBuffer", checkBodyArrayBuffer);
+    checkFetchedResponseBody("formData without correct type", checkBodyFormDataError);
```

Comment:
It might not be very clear for readers that some tests are error cases (meaning we will not be able to get a body) and some are regular cases. Maybe we should group all the error tests together or give them a name suffix like (error case).